rules/utils/cartesian-product-samples.js

```diff
@@ -14,12 +14,16 @@ module.exports = (combinations, length = Infinity) => {
 
 	const samples = Array.from({length: Math.min(total, length)}, (_, sampleIndex) => {
 		let indexRemaining = sampleIndex;
-		return combinations.reduceRight((combination, items) => {
+		const combination = [];
+		for (let i = combinations.length - 1; i >= 0; i--) {
```

Comment:
How to loop forward, if you interest https://github.com/fisker/fast-cartesian-product/blob/master/src/algorithms/calculate-from-index.js#L14